src/components/device/huddlyGrpcTunnelClient.ts

```diff
@@ -0,0 +1,237 @@
+      {
+        send: 'grpc/normal_rpc',
+        receive: 'grpc/normal_rpc_reply',
+      },
+      timeout
+    );
+    return reply;
+  }
+
+  getError({ status_code, status_error_message }: NormalRPCResponse) {
+    if (status_code === 0) {
+      return undefined;
+    }
+    return new GrpcTunnelServiceError(status_error_message, status_code);
+  }
+
+  async runNormalRPCCommand(
+    cmdString: string,
+    request: Uint8Array,
+    callback: Function,
+    deserializer: Function
+  ) {
+    const reply = await this.normalRPC(cmdString, request);
+    const error = this.getError(reply);
+    callback(error, deserializer(reply.response));
+  }
+
+  waitForReady() {
+    return;
+  }
+  close() {
+    this.usbTransport.close();
+  }
+  getDeviceVersion(request, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetDeviceVersion',
+      request.serializeBinary(),
+      callback,
+      huddly.DeviceVersion.deserializeBinary
+    );
+  }
+
+  getLogFiles() {
+    throw new Error('Not Support');
+    return {
+      on: () => {},
+    };
+  }
+
+  eraseLogFile(request: huddly.LogFile, callback: Function) {
+    this.runNormalRPCCommand(
+      'eraseLogFile',
+      request.serializeBinary(),
+      callback,
+      huddly.DeviceStatus.deserializeBinary
+    );
+  }
+
+  reset() {
+    throw new Error('Not Supported.');
+  }
+
+  getCnnFeatureStatus(request: huddly.CnnFeature, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetCnnFeatureStatus',
+      request.serializeBinary(),
+      callback,
+      huddly.CNNStatus.deserializeBinary
+    );
+  }
+
+  getTemperatures(request: Empty, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetTemperatures',
+      request.serializeBinary(),
+      callback,
+      huddly.Temperatures.deserializeBinary
+    );
+  }
+  getBootSlot(request: Empty, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetBootSlot',
+      request.serializeBinary(),
+      callback,
+      huddly.BootSlot.deserializeBinary
+    );
+  }
+  getUptime(request: Empty, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetUptime',
+      request.serializeBinary(),
+      callback,
+      huddly.Uptime.deserializeBinary
+    );
+  }
+
+  getSaturation(request: Empty, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetSaturation',
+      request.serializeBinary(),
+      callback,
+      huddly.Saturation.deserializeBinary
+    );
+  }
+
+  setSaturation(request: huddly.Saturation, callback: Function) {
+    this.runNormalRPCCommand(
+      'SetSaturation',
+      request.serializeBinary(),
+      callback,
+      huddly.DeviceStatus.deserializeBinary
+    );
+  }
+
+  getBrightness(request: Empty, callback: Function) {
+    this.runNormalRPCCommand(
+      'GetBrightness',
+      request.serializeBinary(),
+      callback,
+      huddly.Brightness.deserializeBinary
+    );
+  }
+
+  async setBrightness(request: huddly.Brightness, callback: Function) {
```

Comment:
What's the deal with some functions being async and others not?

Reply:
But to answer your question: it's remnants from the old implementation.